### PR TITLE
Non-google login

### PR DESCRIPTION
**Pull request by b-laci**

### CODE REVIEW

File: service/pom.xml [hunk 1]
```diff
@@ -38,7 +38,12 @@
...
-            <version>1.0</version>
+            <version>${passay.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-core</artifactId>
+            <version>${spring.version}</version>
         </dependency>
     </dependencies>
 </project>
```
Comment (dejanjesic) [inline service/pom.xml]: missing empty line at the end of file

File: service/src/main/java/eu/execom/labs/pomodorospringreact/service/UserServiceImpl.java [hunk 2]
```diff
@@ -283,6 +288,24 @@ public RegistrationUserDto getUserRegistrationForm(final Long invitationId) {
...
+     * User login with email and password.
+     *
+     * @param loginDto LoginDto
+     * @return UserDto
+     */
+    @Transactional(readOnly = true)
+    public UserDto nonGoogleLogin(LoginDto loginDto) {
+        User user = userDao.findByEmail(loginDto.getEmail());
+
+        if (!passwordEncoder.matches(loginDto.getPassword(), user.getPassword())) {
```
Comment (dejanjesic) [inline service/src/main/java/eu/execom/labs/pomodorospringreact/service/UserServiceImpl.java]: Are you matching encoded and non-encoded password here?

File: web/src/main/java/eu/execom/labs/pomodorospringreact/controller/LoginController.java [hunk 3]
```diff
@@ -17,8 +18,13 @@
...
     UserService userService;
 
-    @RequestMapping(method = RequestMethod.POST, value = "")
-    public ResponseEntity<UserDto> login(@RequestBody String encodedData) {
+    @RequestMapping(method = RequestMethod.POST, value = "/google")
+    public ResponseEntity<UserDto> googleLogin(@RequestBody String encodedData) {
         return new ResponseEntity<>(userService.authenticate(encodedData), HttpStatus.OK);
     }
+
+    @RequestMapping(method = RequestMethod.POST, value = "")
```
Comment (dejanjesic) [inline web/src/main/java/eu/execom/labs/pomodorospringreact/controller/LoginController.java]: Why do you need value here?

File: service/src/main/java/eu/execom/labs/pomodorospringreact/service/UserServiceImpl.java (hunk 2 above)
Comment (b-laci) [reply service/src/main/java/eu/execom/labs/pomodorospringreact/service/UserServiceImpl.java]: Yes

Comment (b-laci) [reply service/src/main/java/eu/execom/labs/pomodorospringreact/service/UserServiceImpl.java]: " Verify the encoded password obtained from storage matches the submitted raw password after it too is encoded. Returns true if the passwords match, false if they do not. The stored password itself is never decoded. "

Comment (dejanjesic) [reply service/src/main/java/eu/execom/labs/pomodorospringreact/service/UserServiceImpl.java]: So matches() takes raw password as well as encoded password as parameters, disregard my comment then

**Comment by stanislavgrujic** [comment]
Please resolve merge conflicts.

**Pull request merged**